prime-router/docs/onboarding-users/transport/rest.md

```diff
@@ -0,0 +1,165 @@
+	
+        Command:
+            ./prime credential-create --type UserJks --jks-use <JKS Username> --jks-file-pass <JKS File Password> --jks-file <PATH to JKS File>
+            
+        Output: 
+            Credential in JSON format: 
+            ===
+            {"@type":"UserJks","user":"<JKS Username>", "jks":"<JKS Key>"}
+            ===
+
+<li>Create Azure Key Vault Secret for STLT</li>
+
+With CDC SU password, login into the **Microsoft Azure** and navigate to the **pdhprod-clientconfig** and select **Secrets** from the menu on the left.  You should see the following screenshot.
+
+![](./images/rest-pdhprod-clientconfig-secrets.png)
+
+Next refer to the above image, click on the  [+ Generate/Import] as the red arrow is pointing to.  You should see the following screenshot. Please follow the instruction given in the image.
+
+![](./images/rest-create-a-secrete-key-screenshot.png)
+
+</ol>
+
+## 3.) How to configure the receiver's REST transport
+The receiver's RESTTransport includes the following fields:
+
+    - reportUrl:      The URL to post to. e.g. https://api2.health.ny.gov/services/uphn/V1.0/ECLRSPRE.
+    - authTokenUrl:   The URL to get the OAuth token. e.g. https://api2.health.ny.gov/services/uphn/V1.0/auth.
+    - authType:       Authentication type i.e two-legged.  It is default to null (API shared key)
+    - tlsKeystore:    The name for the credential manager to get the JKS used in TLS/SSL
+    - parameters:     The map of parameters to be sent in the message (use with two-legged OAuth.  See ETOR RESTTransport)
+    - headers:        The map of headers to be sent in the message
+    - type:           "REST"
+
+    a) See UserPass RESTTransport setting Example below:
+        
+        OK-PHD--ELR uses UserPass authentication type:
+        ==============================================
+            transport: !<REST>
+              reportUrl: "https://sample.health.ok.gov/api/document/hl7"
+              authTokenUrl: "https://sample.health.ok.gov/api/auth/token"
+              authType: null
+              tlsKeystore: null
+              parameters: {}
+              headers:
+                senderLabName: "CDC PRIME REPORTSTREAM"
+                RecordId: "header.reportFile.reportId"
+                sourceLabName: "CDC PRIME REPORTSTREAM"
+              type: "REST"
+
+    b) See UserApiKey+JKS TLS/SSL RESTTransport setting Example below:
+
+        NY-PHD--ELR uses UserApiKey + JKS TLS/SSL authentication type:
+        ==============================================================
+            transport: !<REST>
+              reportUrl: "https://samplehealth.ny.gov/services/uphn/openapi/V1.0/ECLRSPOST"
+              authTokenUrl: "https://samplehealth.ny.gov/token"
+              authType: null
+              tlsKeystore: "NY-PHD--ELR--JKS"
```

Comment:
Where is this TLS key store? Is this something we need to set up? Where do we get this value from?

Reply:
TLS is JKS that store in Azure vault where we store other credentials.  Yes to setup we run ./primeCLI as give in create credential step here.
![Screenshot 2023-10-26 at 9 22 22 AM](https://github.com/CDCgov/prime-reportstream/assets/20068283/36d96bd5-f5f8-4eb7-b545-1b164375ac27)